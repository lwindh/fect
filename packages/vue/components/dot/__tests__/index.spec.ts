import { mount } from '@vue/test-utils'
import Dot from '..'

const Wrapper = {
  components: {
    [Dot.name]: Dot
  },
  data() {
    return {
      type: 'default'
    }
  },
  template: `
     <div class="container">
       <fe-dot :type="type" :type-data="type">
        <span>success</span>
       </fe-dot>
     </div>
    `
}

describe('Dot', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Dot)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should support different type', async () => {
    const wrapper = mount(Wrapper)
    await wrapper.setData({ type: 'success' })
    expect(wrapper.find('.fect-dot').attributes('type-data')).toBe('success')
    const el = wrapper.find('.fect-dot__ctx').find('span').text()
    expect(el).toBe('success')
    expect(wrapper.html()).toMatchSnapshot()
    wrapper.unmount()
  })
})
