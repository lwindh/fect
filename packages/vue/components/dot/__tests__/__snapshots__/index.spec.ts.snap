// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dot should be render as a element 1`] = `<div class="fect-dot"><span class="fect-dot__circle fect-dot__circle--default"></span><span class="fect-dot__ctx"><!----></span></div>`;

exports[`Dot should support different type 1`] = `
<div class="container">
  <div class="fect-dot" type-data="success"><span class="fect-dot__circle fect-dot__circle--success"></span><span class="fect-dot__ctx"><span>success</span></span></div>
</div>
`;
